opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -187,17 +147,116 @@ where
 #[derive(Clone)]
 pub struct PeriodicReader {
     exporter: Arc<dyn PushMetricsExporter>,
-    inner: Arc<Mutex<PeriodicReaderInner>>,
+    inner: Arc<PeriodicReaderInner>,
 }
 
 impl PeriodicReader {
     /// Configuration options for a periodic reader
-    pub fn builder<E, RT>(exporter: E, runtime: RT) -> PeriodicReaderBuilder<E, RT>
+    pub fn builder<E>(exporter: E) -> PeriodicReaderBuilder<E>
+    where
+        E: PushMetricsExporter,
+    {
+        PeriodicReaderBuilder::new(exporter)
+    }
+
+    fn new<E>(exporter: E, interval: Duration) -> Self
     where
         E: PushMetricsExporter,
-        RT: Runtime,
     {
-        PeriodicReaderBuilder::new(exporter, runtime)
+        let (message_sender, message_receiver): (Sender<Message>, Receiver<Message>) =
+            mpsc::channel();
+        let reader = PeriodicReader {
+            exporter: Arc::new(exporter),
+            inner: Arc::new(PeriodicReaderInner {
+                message_sender,
+                is_shutdown: AtomicBool::new(false),
+                producer: Mutex::new(None),
+            }),
+        };
+        let cloned_reader = reader.clone();
+
+        thread::spawn(move || {
+            let mut interval_start = Instant::now();
+            let mut remaining_interval = interval;
+            println!("PeriodicReader Thread started.");
+            loop {
+                match message_receiver.recv_timeout(remaining_interval) {
+                    Ok(Message::Flush(response_sender)) => {
+                        println!("Performing ad-hoc export due to Flush.");
+                        if let Err(_e) = cloned_reader.collect_and_export() {
+                            response_sender.send(false).unwrap();
```

Comment:
I am guessing the background worker here probably should not panic if whoever sent it the flush message is no longer there to receive the response - without awareness of any big picture design, my first instinct is to ignore "nobody listening for me anymore" style errors. Perhaps in current code paths this should never happen (did not check all places) but I can easily imagine "don't bother waiting for the flush to complete" a reasonable thing to do at times.